[cmake] remove usage of kodi-platform

olafhering opened the pull request:
The pvr.iptvsimple binary addon does not use kodi-platform.
Remove existing references from CMakeLists.txt and debian/control.

Rechi commented on debian/control:
please fix the syntax error in this line

olafhering commented on debian/control:
Done

olafhering committed:
[cmake] remove usage of kodi-platform

The pvr.iptvsimple binary addon does not use kodi-platform.
Remove existing references from CMakeLists.txt and debian/control.
Add dependency to libp8-platform-dev because libkodiplatform-dev
pulled that dependency in as a side effect. Fix that bug too by
adding an explicit dependency.

Signed-off-by: Olaf Hering <olaf@aepfle.de>
